compiler/block.ts

```diff
@@ -0,0 +1,181 @@
+import * as assert from 'node:assert';
+
+import { RuntimeError } from '../exceptions';
+import { RuntimeFault } from '../faults';
+import {
+  Instr,
+  InstrVisitor,
+  Print,
+  Expression,
+  Rem,
+  New,
+  Load,
+  List,
+  Renum,
+  Save,
+  Run,
+  Exit,
+  End,
+  Let,
+  Assign,
+  ShortIf,
+  If,
+  Else,
+  ElseIf,
+  EndIf,
+} from '../ast/instr';
+
+import { LineCompiler } from './base';
+
+//
+// A class to manage blocks in the LineCompiler. Subclassed for particular
+// kinds of blocks in the base, next to the LineCompiler.
+//
+
+// NOTE: Conceptually, BlockKind is enumerable. However, as the kinds are
+// for the most part defined in the base, setting this to a string allows us
+// to avoid a generic type. In practice, block kinds are only used for error
+// reporting.
+export type BlockKind = string;
+
+export abstract class Block implements InstrVisitor<void> {
```

Comment:
Let's write a test suite for this class. I suspect coverage is OK at the compiler level, but we can still check that `begin`, `next` and `end` work as expected, and boilerplate out the "invalid" commands.